src/main/java/com/cognifide/aemrules/checks/slingquery/SlingQueryImplicitStrategyCheck.java

```diff
@@ -0,0 +1,174 @@
+/*-
+ * #%L
+ * AEM Rules for SonarQube
+ * %%
+ * Copyright (C) 2015 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.checks.slingquery;
+
+import com.cognifide.aemrules.tag.Tags;
+import com.cognifide.aemrules.version.AemVersion;
+import java.util.HashMap;
+import java.util.Map;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.ExpressionStatementTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+@Rule(
+    key = SlingQueryImplicitStrategyCheck.RULE_KEY,
+    name = SlingQueryImplicitStrategyCheck.RULE_MESSAGE,
+    priority = Priority.MINOR,
+    tags = Tags.AEM
+)
+@AemVersion(
+    all = true
+)
+public class SlingQueryImplicitStrategyCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+    protected static final String RULE_KEY = "AEM-19";
+
+    protected static final String RULE_MESSAGE = "Implicit search strategy used in Sling Query";
+
+    private static final String FIND_METHOD = "find";
+
+    private static final String SEARCH_STRATEGY_METHOD = "searchStrategy";
+
+    private static final String SLING_QUERY = "SlingQuery";
```

Comment:
I have a slight concern with these constants, as they provide pretty loose coupling with the actual classes. Can we use fully qualified names instead of using just names? I know it can barely happen that someone will have a class with such name etc. but still it makes me feel uncomfortable.

Reply:
updated